src/main/java/nextstep/courses/domain/course/Generation.java

```diff
@@ -0,0 +1,24 @@
+package nextstep.courses.domain.course;
+
+public class Generation {
+
+  public static final String NEGATIVE_NUMBER_OR_ZERO_IS_NOT_ALLOWED = "기수 값은 0이나 음수는 허용되지 않습니다. input: %s";
+
+  private Long id;
+
+  public Generation(Long id) {
+    validate(id);
+    this.id = id;
+  }
+
+  public boolean isSame(Long id) {
+    return this.id.equals(id);
+  }
+
+  private boolean validate(Long number) {
+    if (number <= 0) {
+      throw new IllegalArgumentException(String.format(NEGATIVE_NUMBER_OR_ZERO_IS_NOT_ALLOWED, number));
+    }
+    return true;
+  }
```

Comment:
```suggestion
  private void validate(Long number) {
    if (number <= 0) {
      throw new IllegalArgumentException(String.format(NEGATIVE_NUMBER_OR_ZERO_IS_NOT_ALLOWED, number));
    }
  }
```
반환타입은 필요 없어보입니다.
void 타입이 적절하지 않을까요?

그리고 생성자에서 사용하는 private 메소드이니 생성자 아래에 배치하면 좋겠습니다.

Reply:
준혁님 말씀주셔서 감사해요!
저는 private 메서드는 public보다 늘 뒤에 둬야하는지 알았습니다!
확실히 생성자 밑에 있는 게 가독성이 더 좋은 거 같아요!
알려주셔서 감사합니다~~